web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -96,13 +88,18 @@ public function blockForm($form, FormStateInterface $form_state) {
       '#default_value' => $this->configuration['cities'] ?? reset($cities),
       '#description' => $this->t('Select the city.'),
     ];
-    $form['API - key'] = [
+    $form['settings']['key'] = [
+      '#required' => TRUE,
       '#type' => 'textfield',
       '#title' => $this->t('Your API key'),
       '#default_value' => $this->configuration['key'] ?? 'Your API Key',
       '#description' => $this->t('Enter your API key'),
     ];
 
+    if (!empty($this->configuration['settings']['key'])) {
+      $form['settings']['key']['#default_value'] = $this->configuration['key'];
```

Comment:
`'#default_value' => $this->configuration['key'] ?? NULL`
directly in the array above